### PR TITLE
Speed up tests by not type-checking lib.d.ts unless the test asks for that.

### PR DESCRIPTION
**Pull request by CyrusNajmabadi**

### CODE REVIEW

File: src/compiler/types.ts [hunk 1]
```diff
@@ -1823,6 +1824,7 @@ module ts {
         experimentalDecorators?: boolean;
         emitDecoratorMetadata?: boolean;
         /* @internal */ stripInternal?: boolean;
+        /* @internal */ skipDefaultLibCheck?: boolean;
```
Comment (mhegazy) [inline src/compiler/types.ts]: why not `noLibCheck`?

File: src/compiler/checker.ts [hunk 2]
```diff
@@ -11368,6 +11368,12 @@ module ts {
         function checkSourceFileWorker(node: SourceFile) {
             let links = getNodeLinks(node);
             if (!(links.flags & NodeCheckFlags.TypeChecked)) {
+                links.flags |= NodeCheckFlags.TypeChecked;
```
Comment (mhegazy) [inline src/compiler/checker.ts]: why mark it if we are skipping it? the two checks seem to be more or less the same cost, but it is misleading.

File: src/compiler/types.ts (hunk 1 above)
Comment (mhegazy) [inline src/compiler/types.ts]: We also should expose this as a commandline option.

File: src/compiler/checker.ts (hunk 2 above)
Comment (CyrusNajmabadi) [inline src/compiler/checker.ts]: Sure.  will revert this.

**Pull request merged**